third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -1628,6 +1628,24 @@ struct LoadOpConversion
     numOperandsInnerDimPerLoad =
         isOperandA ? numOperandsPer2DloadN : numOperandsPer2DLoadM;
 
+    // downscale if the load size is bigger than the block size
+    LLVM_DEBUG({
+      llvm::dbgs() << "numOperandsOuterDimPerLoad before downscaling = "
+                   << numOperandsOuterDimPerLoad << "\n";
+      llvm::dbgs() << "numOperandsInnerDimPerLoad before downscaling = "
+                   << numOperandsInnerDimPerLoad << "\n";
+    });
+    numOperandsOuterDimPerLoad =
+        std::max(std::min(numOperandsOuterDimPerLoad,
+                          static_cast<unsigned>(tensorShape[dimOuter] /
+                                                elemsPerDPASInst[0])),
+                 1u);
```

Comment:
Is the max(x, 1) to avoid the case where `tensorShape[dimOuter] % elemsPerDPASInst[0] != 0`? If so, maybe safer to do a ceil like below?
```suggestion
        std::min(numOperandsOuterDimPerLoad, static_cast<unsigned>(tensorShape[dimOuter] / elemsPerDPASInst[0] + (tensorShape[dimOuter] % elemsPerDPASInst[0] != 0)))
```

Reply:
The max(x, 1) is in place for the case where the tensor shape (block size) is smaller than a DPAS tile. In that case, we would get 0 from `tensorShape[dimOuter] / elemsPerDPASInst[0]` which would be incorrect - we always need at least one outer/inner load operand per DPAS instruction currently.

Reply:
When `tensorShape[dimOuter] % elemsPerDPASInst[0] != 0`, e.g., `tensorShape[dimOuter] = 3` and `elemsPerDPASInst[0] = 2`, do we want `numOperandsOuterDimPerLoad` to be `1` or `2`?

Reply:
Ah, good point - I do not know if that can happen or not. `elemsPerDPASInst` is basically either 8, 16, or in one case 32 - so we would need a non-power of 2 tensor shape which I do not think is allowed. However, I think the utility function `mlir::ceil` does what we want so I switched to that in the latest commit.